presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetCompressionUtils.java

```diff
@@ -45,23 +47,22 @@ public static Slice decompress(CompressionCodecName codec, Slice input, int unco
             case GZIP:
                 return decompressGzip(input, uncompressedSize);
             case SNAPPY:
-                return decompressSnappy(input, uncompressedSize);
+                return decompress(new SnappyDecompressor(), input, uncompressedSize);
+            case LZO:
+                return decompress(new LzoDecompressor(), input, uncompressedSize);
```

Comment:
So I wrote a new test that shows how to decompress using the Hadoop framing on the LZO algorithm: https://gist.github.com/dain/dd2032793f16ab87e5b4500ef6b77a7f

There are two versions, the first simply using the HadoopLzoInputStream in Aircompressor.  This is simple but I don't think it is as efficient since it must manage a stream.  In the second version, I implemented your decompressor method using code extracted from the HadoopLzoInputStream to decompress directly into an output buffer.  This should be more efficient, but it should be reviewed carefully (especially because I did this very very quickly and it is very late).

It would also be helpful to dig into the Parquet code to verify they the exact code they are using to compress LZO data (and the framing format they are using).